framework/test/jest/unit/specs/modules/chain/synchronizer/block_synchronization_mechanism.spec.js

```diff
@@ -0,0 +1,183 @@
+const BlockSynchronizationMechanism = require('../../../../../../../src/modules/chain/synchronizer/block_synchronization_mechanism');
+
+describe('BlockSynchronizationMechanism', () => {
+	const stubs = {
+		channel: {
+			invoke: jest.fn(),
+		},
+		modules: {
+			blocks: {
+				lastBlock: jest.fn(),
+			},
+		},
+	};
+	let blockSynchronizationMechanism;
+
+	beforeEach(() => {
+		blockSynchronizationMechanism = new BlockSynchronizationMechanism({
+			channel: stubs.channel,
+			modules: stubs.modules,
+		});
+	});
+
+	describe('constructor()', () => {});
+
+	describe('_computeBestPeer()', () => {
+		it('should accept an array of peers as input', () => {
+			expect(blockSynchronizationMechanism._computeBestPeer.length).toEqual(1);
+		});
+
+		it('should return a peer object', () => {
+			stubs.modules.blocks.lastBlock = {
+				height: 0,
+				prevotedConfirmedUptoHeight: 0,
+			};
+
+			const peers = [
+				{
+					lastBlockId: '12343245',
+					prevotedConfirmedUptoHeight: 1,
+					height: 67,
+					ip: '127.0.0.1',
+				},
+			];
+
+			expect(blockSynchronizationMechanism._computeBestPeer(peers)).toEqual(
+				peers[0]
+			);
+		});
+
+		describe('given a set of peers', () => {
+			/**
+			 * a) Among all peers, choose the subset of peers with largest T.prevotedConfirmedUptoHeight.
+			 * b) From the peers in a) choose those with largest T.height.
+			 * c) Form the peers in b) choose the largest set which has the same blockID.
+			 * 		Ties are broken in favor of smaller blockID.
+			 *
+			 * This set is the set bestPeers and we define B to be the block at the
+			 * tip of the chain of this set of peers.
+			 *
+			 * @link https://github.com/LiskHQ/lips/blob/master/proposals/lip-0014.md#block-synchronization-mechanism
+			 */
+
+			it('should successfully select the best peer according to the steps defined in LIP-0014', () => {
```

Comment:
Does it make sense to have a similar structure of CSV file for test scenarios in selecting best peers?

Reply:
I think this case is different as the volume of data and input is way less than in BFT case and so it's easier to read as a JSON object. I agree with you that perhaps I can add more tests cases to cover more scenarios.